packages/gasket-core/lib/engine.js

```diff
@@ -12,7 +14,7 @@ class GasketEngine {
 
     this._hooks = {};
     this._plans = {};
-    this._traceDepth = 0;
+    this._traceStack = [];
```

Comment:
I am realizing that this isn't exactly safe.

```
import gasket from './gasket.js';

const worker1 = gasket.exec('do-something-that-takes awhile');
const worker2 = gasket.exec('do-something-that-takes awhile');

await Promise.all([worker1, worker2]);
```

This will increase the trace depth. While not a common scenario, it is bound to occur when handling multiple requests.

I'll add some more tests for now and cycle back to this.

Reply:
Accounting for this scenario accurately ended up snowballing into a decent amount. I'm requesting re-reviews since of the PR as a result.